provider/provider_config/k8s_config.go

```diff
@@ -22,24 +23,94 @@ func (k8s *K8sConfig) Serialize() ([]byte, error) {
 	return data, nil
 }
 
-func (k8s *K8sConfig) Deserialize(config SerializedConfig) error {
-	err := json.Unmarshal(config, k8s)
+func (k8s *K8sConfig) Deserialize(data SerializedConfig) error {
+	err := json.Unmarshal(data, k8s)
 	if err != nil {
-		return fmt.Errorf("deserialize k8s config: %w", err)
+		return err
 	}
-	if k8s.ExecutorConfig == "" {
+	return nil
+}
+
+func (k8s *K8sConfig) UnmarshalJSON(data []byte) error {
+	type tempConfig struct {
+		ExecutorType   ExecutorType
+		ExecutorConfig interface{}
+		StoreType      FileStoreType
+		StoreConfig    map[string]interface{}
+	}
+
+	var temp tempConfig
+	err := json.Unmarshal(data, &temp)
+	if err != nil {
+		return fmt.Errorf("unmarshal: %w", err)
```

Comment:
I'm wondering if we should make this error message more specific compared to the other error messages. Something along the lines of `"failed to unmarshal configuration: %v"`.